### ISSUE TITLE
Investigate BuildStream as part of the dependable build steps of the ODP

### ISSUE THREAD

**Issue by adamj-codethink** [finos/open-developer-platform#168]
### Description of Problem:
The ODP project tries to provide recommendations for secure/compliant software.

I feel when a commit is made to a FinOS project, we could build the software in a fully sandboxed environment, which could produce a container/runtime that will allow us to verify the software meets the expected standards/compliance.

### Potential Solutions:
One solution to this would be to use [BuildStream](https://gitlab.com/BuildStream/buildstream), which is a Free Software tool for
building/integrating software stacks.

This is an Free Software tool, which is hosted under the Apache Foundation.

As part of this investigation into whether the ODP group feel this tool could help with this part of the ODP process, we could provide some form of demo/presentation in the future, if this would be of interest?

**Comment by maoo** [comment]
Love the idea! BuildStream sounds like a very promising solution to have an holistic view on build and release activities of FINOS hosted projects!

In terms of requirements, as we briefly discussed during today's ODP call, I think it would be great if all FINOS GitHub repositories expose a badge (automatically posted from a PR) that reflects:
  - If the repo has one or more builds
  - If the builds work
  - What are the build systems used
  - What are the latest artifact versions
  - Where to download artifacts

I'm happy to support this activity and use ODP as a venue for discussion.

Thank you @adamj-codethink !

**Comment by adamj-codethink** [comment]
Thanks @maoo 

I agree, i think as part of ODP, it would be good for us to promote these ideas to all projects.

I know BuildStream could help with most of the points, but i would be interested to know if people have used any other tools also.

Where is the CI/CD hosted at the moment, do you know? We could maybe start by just showing the latest build as a badge, then create a separate ticket for the other bullet points?

**Comment by maoo** [comment]
> Where is the CI/CD hosted at the moment, do you know? We could maybe start by just showing the latest build as a badge, then create a separate ticket for the other bullet points?

There is single CI/CD tool at FINOS, we provide different options, see [odp.finos.org/docs/development-infrastructure/continuous-integration/intro](https://odp.finos.org/docs/development-infrastructure/continuous-integration/intro); we see that FINOS community is conveying on GitHub Actions, so we started using it internally to provide some automation to projects, see [github.com/finos/open-developer-platform/tree/master/.github/workflows](https://github.com/finos/open-developer-platform/tree/master/.github/workflows) . Projects like [DataHub](https://github.com/finos/datahub/actions?query=workflow%3Adatabub-build) are also using it to run code builds.

Another challenge for us is to enforce this action across all our repos, with the ability to roll it out gradually; a GitHub App would provide an easier way to enable this feature across all repos of a given GitHub org; that's what we have experienced when we helped building and deployed the [CLA Bot](http://finos.github.io/cla-bot).

Maybe we could start from GitHub Actions as a point of integration, given that they're very simple to write and test, and then evaluate the GitHub App option?
